### ISSUE TITLE
1.3.x: Trend arrow missing from current BG value (in header/dashboard)

### ISSUE THREAD

**Issue by danibjor** [LoopKit/Loop#456]
Looks like the trend arrow is missing from the BG value in the header on the newer builds. I think it's v1.3.x+

**Comment by ps2** [comment]
What CGM are you using?

**Comment by danibjor** [comment]
Enlite. It did work before. Tried with the latest Dev bits from approx 24 hours back, and it's still missing

**Comment by ps2** [comment]
I don't believe we parse arrows on enlite on x22.  Are you x23?  Have you done a mysentry pairing?

**Comment by danibjor** [comment]
My bad. Should know better than not including basic stuff when reporting issues..

754 EU pump

That's running pre-1.3 (with trend arrow)

![unnamed](https://cloud.githubusercontent.com/assets/2149381/25422231/86a44076-2a60-11e7-9da9-e0d2f7d06aaf.jpg)

**Comment by ps2** [comment]
Yes, I think I see what's going on. It was part of the CGM refactor that happened in this version.  We track a variable called latestPumpStatusFromMySentry that contains trend from mysentry packets.  We still track that var, but the displayed trend arrow comes from sensorState on the cgm data manager (EnliteCGMManager).  I think we need to update the instance of EnliteCGMManager with a new sensorState when a mysentry packet comes in, and also make sure to not overwrite it when history cgm data is fetched, unless the history cgm value is newer.

I don't run Enlite, so this would be a hard change for me to test.  Are you able to attempt this change and submit a PR?

**Comment by danibjor** [comment]
I'll take a look and see if I could debug this - and maybe provide a fix. 

Any tips are welcome as I don't have the full overview after the stuff was refactored.

**Comment by iMaces** [comment]
Any kind of update on this? I would love the trend arrows to be back for elite users. (Still not working on latest dev build). Would like to help but don't know where to begin (very new to coding).
